src/Http/ReceivedAuthnRequestQueryString.php

```diff
@@ -26,59 +26,37 @@
 
 final class ReceivedAuthnRequestQueryString implements SignatureVerifiable
 {
-    const PARAMETER_REQUEST = 'SAMLRequest';
-    const PARAMETER_SIGNATURE = 'Signature';
-    const PARAMETER_SIGNATURE_ALGORITHM = 'SigAlg';
-    const PARAMETER_RELAY_STATE = 'RelayState';
+    public const PARAMETER_REQUEST = 'SAMLRequest';
+    public const PARAMETER_SIGNATURE = 'Signature';
+    public const PARAMETER_SIGNATURE_ALGORITHM = 'SigAlg';
+    public const PARAMETER_RELAY_STATE = 'RelayState';
 
-    private static $samlParameters = [
+    private static array $samlParameters = [
         self::PARAMETER_REQUEST,
         self::PARAMETER_SIGNATURE,
         self::PARAMETER_SIGNATURE_ALGORITHM,
         self::PARAMETER_RELAY_STATE,
     ];
 
-    /**
-     * @var string
-     */
-    private $samlRequest;
+    private string $samlRequest;
 
-    /**
-     * @var string|null
-     */
-    private $signature;
+    private ?string $signature = null;
 
-    /**
-     * @var string|null
-     */
-    private $signatureAlgorithm;
+    private ?string $signatureAlgorithm = null;
 
-    /**
-     * @var string|null
-     */
-    private $relayState;
+    private ?string $relayState = null;
 
     private function __construct($samlRequest)
     {
         $this->samlRequest = $samlRequest;
     }
 
     /**
-     * @param string $query
-     * @return ReceivedAuthnRequestQueryString
-     *
      * @SuppressWarnings(PHPMD.CyclomaticComplexity) Extensive validation
      * @SuppressWarnings(PHPMD.NPathComplexity) Extensive validation
      */
-    public static function parse($query)
+    public static function parse(string $query): ReceivedAuthnRequestQueryString
     {
-        if (!is_string($query)) {
-            throw new InvalidReceivedAuthnRequestQueryStringException(sprintf(
```

Comment:
This is quite a specific exception.. Did you make sure it wasn't caught anywhere, because a TypeError is an Error instead of an Exception and would potentially not get caught.

Reply:
Also, some of the exceptions in the library may have gotten redundant. It may be worth checking if they are still in use.